internal/worker/firewaller/firewaller.go

```diff
@@ -485,8 +500,12 @@ func (fw *Firewaller) startMachine(ctx context.Context, tag names.MachineTag) er
 		if !ok {
 			return errors.New("machine units watcher closed")
 		}
+		unitNames, err := transform.SliceOrErr(change, coreunit.NewName)
```

Comment:
Not sure I'd bother with this extra step. Just make it a string. We will make the watcher emit UUIDs later anyway.